### PR TITLE
Speedup GBRTree by making it non branching 

### PR DESCRIPTION
**Pull request by VinInn**
The code navigating a GBR Tree has been made explicitly not branching as the compiler (actually gcc) stubbornly produces branching code.

see
https://godbolt.org/z/aGz4eTad3
(yes it seems that gcc13 produces a different assembler closer to the bare C++. Apparently it has been fixed in the trunk)
(and yes, clang generates the same code for the old and new source)

code tested in HLT where SeedMVAEstimator is a non negligible contributor
SeedMVAEstimator speedups by itself of 40%.
The caller module hltIter0IterL3FromL1MuonPixelSeedsFromPixelTracksFiltered by a similar amount.

regression may appear due to the changes in SeedMVAEstimator itself (double ->float, div -> prod)

### CODE REVIEW

**Comment by VinInn** [comment]
@cmsbuild : please test

**Comment by mandrenguyen** [comment]
please test

File: RecoMuon/TrackerSeedGenerator/src/SeedMvaEstimator.cc [hunk 1]
```diff
@@ -39,19 +42,18 @@ void SeedMvaEstimator::getL1MuonVariables(const GlobalVector& global_p,
...
-
-      float dR2tmp = reco::deltaR2(it->etaAtVtx(), it->phiAtVtx(), global_p.eta(), global_p.phi());
-      if (dR2tmp < dR2dRL1SeedP) {
-        dR2dRL1SeedP = dR2tmp;
-        dPhidRL1SeedP = reco::deltaPhi(it->phiAtVtx(), global_p.phi());
-      }
+  int ibx = 0;  // -- only take when ibx == 0 -- //
+
+  auto eta = global_p.eta();
+  auto phi = global_p.barePhi();
```
Comment (jfernan2) [inline RecoMuon/TrackerSeedGenerator/src/SeedMvaEstimator.cc:48]: Is there any implication in moving from phi() to barePhi() ?

Comment (VinInn) [reply RecoMuon/TrackerSeedGenerator/src/SeedMvaEstimator.cc:48]: absolutely equivalent in this contest

Comment (VinInn) [reply RecoMuon/TrackerSeedGenerator/src/SeedMvaEstimator.cc:48]: see https://cmssdt.cern.ch/lxr/source/DataFormats/Math/interface/deltaR.h#0036

**Comment by jfernan2** [comment]
+1

**Comment by perrotta** [comment]
+1
- Kind of magic for me, but I trust what I see in the compiler explorer...

**Comment by antoniovilela** [comment]
+1

**Comment by mmusich** [comment]
type performance-improvements

**Pull request merged**